sympy/polys/matrices/linsolve.py

```diff
@@ -158,25 +158,64 @@ def _linear_eq_to_dict(eqs, syms):
 
 
 def _linear_eq_to_dict_inner(eqs, syms):
-    """Convert a system Expr/Eq equations into dict form"""
+    """Convert a system Expr/Eq equations into dict form, returning
+    the coefficient dictionaries and a list of syms-independent terms
+    from each expression in ``eqs```.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.matrices.linsolve import _linear_eq_to_dict_inner as F
+    >>> from sympy.abc import x
+    >>> F([2*x + 3], {x})
+    ([{x: 2}], [3])
+    """
     syms = set(syms)
-    eqsdict, eqs_rhs = [], []
+    eqsdict, ind = [], []
     for eq in eqs:
-        rhs, eqdict = _lin_eq2dict(eq, syms)
+        c, eqdict = _lin_eq2dict(eq, syms)
         eqsdict.append(eqdict)
-        eqs_rhs.append(rhs)
-    return eqsdict, eqs_rhs
+        ind.append(c)
+    return eqsdict, ind
 
 
-def _lin_eq2dict(a, symset):
-    """Efficiently convert a linear equation to a dict of coefficients"""
+def _lin_eq2dict(a, symset, strict=True):
+    """return (c, d) where c is the sym-independent part of ``a`` and
+    ``d`` is an efficiently calculated dictionary mapping symbols to
+    their coefficients. A PolyNonlinearError is raised if non-linearity
+    is detected. To allow cross-terms involving object containing (but not
+    equal to) a symbol, use ``strict=False``
+
```

Comment:
```suggestion

    The values in the dictionary will be non-zero if the original expression was expanded
    but may contain expressions which will simplify to zero, otherwise.
```